src/main/java/com/iota/iri/utils/log/StatusLogger.java

```diff
@@ -0,0 +1,133 @@
+     *
+     * When creating the {@link Thread} it copies its name so the output is "transparent" to the user and the effect
+     * is the same as dumping the message manually through the {@link Logger} object itself.
+     *
+     * @param message status message that shall get printed
+     */
+    public void status(String message) {
+        if (!message.equals(statusMessage)) {
+            statusMessage = message;
+
+            if (System.currentTimeMillis() - lastLogTime >= LOG_INTERVAL) {
+                printStatusMessage();
+            } else if (outputThread == null) {
+                synchronized(this) {
+                    if (outputThread == null) {
+                        outputThread = new Thread(() -> {
+                            try {
+                                Thread.sleep(Math.max(LOG_INTERVAL - (System.currentTimeMillis() - lastLogTime), 1));
+                            } catch(InterruptedException e) { /* do nothing */ }
+
+                            printStatusMessage();
+
+                            outputThread = null;
+                        }, Thread.currentThread().getName());
+
+                        outputThread.start();
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * This method issues an error message.
+     *
+     * Error messages will always get dumped immediately instead of scheduling their output. Since error messages are
+     * usually a sign of some misbehaviour of the node we want to be able to see them when they appear (to be able to
+     * track down bugs more easily).
+     *
+     * @param errorMessage error message that shall get dumped
+     */
+    public void error(String errorMessage) {
+        logger.error(errorMessage);
+    }
+
+    /**
+     * This method issues an error message and also prints the stack trace of the given exception.
+     *
+     * Error messages will always get dumped immediately instead of scheduling their output. Since error messages are
+     * usually a sign of some misbehaviour of the node we want to be able to see them when they appear (to be able to
+     * track down bugs more easily).
+     *
+     * @param errorMessage error message that shall get dumped
+     * @param cause exception object that caused the error to happen (will be printed with the complete stacktrace)
+     */
+    public void error(String errorMessage, Throwable cause) {
+        logger.error(errorMessage, cause);
+    }
+
+    /**
+     * This method handles the actual output of messages through the {@link Logger} instance.
+     *
+     * It first checks if the message that shall get printed differs from the last message that was printed and then
+     * issues the output of the message. After printing the message, it updates the internal variables to handle the
+     * next message accordingly.
+     */
+    private void printStatusMessage() {
+        if (!statusMessage.equals(lastPrintedStatusMessage)) {
```

Comment:
You are already checking in the `status` method if the message you just received was equal to the previous one.

Reply:
Imagine the followign example:

.status("A") -> gets printed
.status("B") -> gets scheduled
.status("A") -> would get printed again even tho we already showed this exact status.

Reply:
Oh I see: it's last received vs last printed.